### ISSUE TITLE
[Train] Change dataset key from "val" to "evaluation".

### ISSUE THREAD

**Issue by woshiyyya** [ray-project/ray#36873]
### Description

Currently LightningTrainer is using "val" as a special key for validation set. We actually have a constant `EVALUATION_DATASET_KEY` in `ray.air.constants`, it's plausible to use this constant for all trainers.

**Comment by woshiyyya** [comment]
Close it since we've deprecated `LightningTrainer` and migrated to `TorchTrainer`.
